Altis_Life.Altis/core/housing/fn_houseMenu.sqf

```diff
@@ -92,14 +92,14 @@ if (!(_curTarget in life_vehicles) || isNil {_curTarget getVariable "house_owner
 
     private _isHouse = false;
 
-    if (isClass (missionConfigFile >> "Housing" >> worldName >> (typeOf _curTarget))) then {
-        _Btn1 ctrlSetText localize "STR_pInAct_BuyHouse";
-        _Btn1 buttonSetAction "[life_pInact_curTarget] spawn life_fnc_buyHouse;";
-        _isHouse = true;
-    } else {
-        _Btn1 ctrlSetText localize "STR_pInAct_BuyGarage";
-        _Btn1 buttonSetAction "[life_pInact_curTarget] spawn life_fnc_buyHouse;";
-    };
+    private _buildingPurchaseString = [
+        "STR_pInAct_BuyGarage",
+        "STR_pInAct_BuyHouse"
+    ] select (isClass (missionConfigFile >> "Housing" >> worldName >> typeOf _curTarget));
+    
+    _isHouse = (isClass (missionConfigFile >> "Housing" >> worldName >> typeOf _curTarget));
```

Comment:
I'd recommend moving this before line 95, and you can use `_isHouse` in your string selection